src/coreclr/tools/Common/System/Collections/Generic/ArrayBuilder.cs

```diff
@@ -29,6 +31,19 @@ public void Add(T item)
             _items[_count++] = item;
         }
 
+#if NET
+        public readonly Span<T> AsSpan(int start) => _items.AsSpan(start);
```

Comment:
Shouldn't this call limit the span to the used range using count, instead of returning the total remaining span?

Reply:
Good catch! That is one way to think of methods like this and that's what the `ValueStringBuilder` does. I didn't think of this because I looked at how the existing indexer allows and it doesn't validate for this (you index into and mutate the "remaining span" while `ValueStringBuilder` has `Debug.Assert` to disallow this.

@jkotas thoughts on whether we should follow VSB and add assert to the indexer & change the slice returned here to the already written portion?

Reply:
I do not have opinions on this. These internal helpers take number of shortcuts that would not be appropriate for public APIs. This is one of them.